From 89982c7818b9c69935aadd09ff9ab09979754cb9 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 20 Sep 2024 17:21:57 +0200
Subject: [PATCH] dt-bindings: stm32: add c3 clock binding for Cortex-CM0 on
 stm32mp25 platform

Add CPU3 clock binding entry for STM32MP25 SoC family.

Change-Id: I3c3e27dfd2ac60e7635588d3fc974e8296b3efaf
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/403707
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 .../bindings/clock/st,stm32mp25-rcc.yaml      |  8 +++-
 include/dt-bindings/clock/st,stm32mp25-rcc.h  | 39 ++++++++++++++++++-
 2 files changed, 44 insertions(+), 3 deletions(-)

--- a/Documentation/devicetree/bindings/clock/st,stm32mp25-rcc.yaml
+++ b/Documentation/devicetree/bindings/clock/st,stm32mp25-rcc.yaml
@@ -118,6 +118,9 @@ properties:
       - description: CK_SCMI_TIMG2 Peripheral bridge for timer2
       - description: CK_SCMI_PLL3 PLL3 clock
       - description: clk_dsi_txbyte DSI byte clock
+      - description: CK_SCMI_HSI_KER_CK Kernel High Speed Internal oscillator
+      - description: CK_SCMI_HSE_KER_CK Kernel High Speed External oscillator
+      - description: CK_SCMI_MSI_KER_CK Kernel Low Power Internal oscillator
 required:
   - compatible
   - reg
@@ -215,6 +218,9 @@ examples:
                   <&scmi_clk CK_SCMI_TIMG1>,
                   <&scmi_clk CK_SCMI_TIMG2>,
                   <&scmi_clk CK_SCMI_PLL3>,
-                  <&clk_dsi_txbyte>;
+                  <&clk_dsi_txbyte>,
+                  <&scmi_clk CK_SCMI_HSI_KER_CK>,
+                  <&scmi_clk CK_SCMI_HSE_KER_CK>,
+                  <&scmi_clk CK_SCMI_MSI_KER_CK>;
     };
 ...
--- a/include/dt-bindings/clock/st,stm32mp25-rcc.h
+++ b/include/dt-bindings/clock/st,stm32mp25-rcc.h
@@ -365,9 +365,37 @@
 #define HSE_DIV2_CK		343
 #define CK_KER_ETR		344
 #define CK_KER_STM		345
-#define CK_CPU3			346
+#define HSI_KER_CK		346
+#define HSE_KER_CK		347
+#define MSI_KER_CK		348
+#define CK_CPU3			349
+#define CK_ADF1_C3		350
+#define CK_GPIOZ_C3		351
+#define CK_I2C8_C3		352
+#define CK_I3C4_C3		353
+#define CK_LPDMA_C3		354
+#define CK_LPTIM3_C3		355
+#define CK_LPTIM4_C3		356
+#define CK_LPTIM5_C3		357
+#define CK_LPUART1_C3		358
+#define CK_RTC_C3		359
+#define CK_SPI8_C3		360
+#define CK_ADF1_AM		361
+#define CK_CPU3_AM		362
+#define CK_GPIOZ_AM		363
+#define CK_I2C8_AM		364
+#define CK_I3C4_AM		365
+#define CK_IPCC2_AM		366
+#define CK_LPDMA_AM		367
+#define CK_LPTIM3_AM		368
+#define CK_LPTIM4_AM		369
+#define CK_LPTIM5_AM		370
+#define CK_LPUART1_AM		371
+#define CK_RTC_AM		372
+#define CK_SPI8_AM		373
+
+#define STM32MP25_LAST_CLK	374
 
-#define STM32MP25_LAST_CLK	347
 
 #define CK_SCMI_ICN_HS_MCU	0
 #define CK_SCMI_ICN_SDMMC	1
@@ -493,5 +521,12 @@
 #define CK_SCMI_BUS_STM		122
 #define CK_SCMI_KER_STM		123
 #define CK_SCMI_KER_ETR		124
+#define CK_SCMI_HSI_KER_CK	125
+#define CK_SCMI_HSE_KER_CK	126
+#define CK_SCMI_MSI_KER_CK	127
+#define CK_SCMI_GPIOZ_AM	128
+#define CK_SCMI_IPCC2_AM	129
+#define CK_SCMI_LPDMA_AM	130
+#define CK_SCMI_RTC_AM		131
 
 #endif /* _DT_BINDINGS_STM32MP25_CLKS_H_ */
